### ISSUE TITLE
Sa moara viata mea

### ISSUE THREAD

**Issue by CodingRule** [Otincelescud/Conway-s-Game-of-Life#2]
Pai sa mor eu de nu merge jocu te rog eu sa moara veata mea otincelescusus sa repari jocu vietii mele

**Comment by CodingRule** [comment]
@Otincelescud

**Comment by 405oops** [comment]
Certainly, I can provide some suggestions for improving the code. Here are a few areas where you can make enhancements:

    Code Organization:
        Consider encapsulating the code within functions or classes for better organization and readability.

    Magic Numbers:
        Replace "magic numbers" like 30 and 2 with named constants to make the code more self-explanatory.

    Comments:
        Add comments to explain the purpose of the code blocks, functions, and any complex logic. This will help others understand your code.

    Grid Initialization:
        It's a good practice to separate the grid initialization from the main function. You can create a separate function to initialize the grid.

    Constants for Colors:
        Define constants for colors to make the code more readable. For example, (255, 255, 255) and (6, 7, 66) can be replaced with named constants.

    Input Handling:
        Handle input more gracefully, especially mouse input, by abstracting it into functions or classes to improve readability and maintainability.

    Avoid Global Variables:
        Minimize the use of global variables by encapsulating them within classes or functions where appropriate.

    Use Meaningful Variable Names:
        Choose variable names that are more descriptive. For instance, toggle_to can be named something like current_state.

    Performance Optimization:
        Depending on the size of the grid, you can explore optimization techniques to improve performance if needed.

    Error Handling:
        Implement error handling to handle exceptional cases gracefully.

Here's an example of how you might encapsulate code within a class:

python

import pygame

class ConwayGame:
    def __init__(self, width, height, px_per_cell, border):
        # Initialize pygame and other attributes

    def create_grid(self):
        # Initialize the grid here

    def get_neighbours_pos(self, pos):
        # Implement neighbor calculation

    def is_in_grid(self, pos):
        # Implement boundary checking

    def display_cell(self, alive, pos):
        # Implement cell display

    def run_simulation(self):
        # Main game loop with the simulation logic

    def handle_input(self):
        # Handle input events

    def main(self):
        # Initialize the game and run the main loop

if __name__ == "__main__":
    game = ConwayGame(800, 800, 10, 1)
    game.main()

These are just some suggestions to improve code structure and readability. You can further enhance the code based on your specific requirements and preferences.

**Comment by CodingRule** [comment]
Bagamias pl in viata vietii @Otincelescud
